File: app/src/main/java/com/wafflestudio/snutt/adapter/NotificationAdapter.java
package com.wafflestudio.snutt.adapter;

import android.app.Activity;
import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;
import android.widget.TextView;

import com.wafflestudio.snutt.R;
import com.wafflestudio.snutt.model.Notification;
import com.wafflestudio.snutt.model.SettingsItem;

import java.util.List;

/**
 * Created by makesource on 2017. 2. 27..
 */

public class NotificationAdapter  extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private static final String TAG = "NOTIFICATION_ADAPTER";
    private List<Notification> lists;

    public NotificationAdapter(List<Notification> lists) {
        this.lists = lists;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.cell_notification, parent, false);
        //View view = inflater.inflate(R.layout.cell_notification, null);
        return new NotificationViewHolder(view);
    }


    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        //final int itemType = getItemViewType(position);
        ((NotificationViewHolder)holder).bindData(getItem(position));
    }

    public Notification getItem(int position) {
        return lists.get(position);
    }

    @Override
    public int getItemViewType(int position) {
        //SettingsItem item = getItem(position);
        return 0;
    }

    @Override
    public int getItemCount() {
        Log.d(TAG, "notification list size : " + lists.size());
        return lists.size();
    }

    // inner class to hold a reference to each item of RecyclerView
    private static class NotificationViewHolder extends RecyclerView.ViewHolder {
        private TextView message;

        private NotificationViewHolder(View view) {
            super(view);
            message = (TextView) view.findViewById(R.id.message);
        }

        private void bindData(Notification notification) {
            Log.d(TAG, "notification message : " + notification.getMessage());
            message.setText(notification.getMessage());
        }
    }
}

File: app/src/main/java/com/wafflestudio/snutt2/views/logged_in/notification/NotificationAdapter.kt
package com.wafflestudio.snutt2.views.logged_in.notification

import android.content.res.Resources
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.compose.ui.res.stringResource
import androidx.paging.PagingDataAdapter
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.RecyclerView
import com.wafflestudio.snutt2.R
import com.wafflestudio.snutt2.databinding.ItemNotificationBinding
import com.wafflestudio.snutt2.lib.base.BaseAdapter
import com.wafflestudio.snutt2.lib.base.BaseViewHolder
import com.wafflestudio.snutt2.lib.network.dto.core.NotificationDto

class NotificationAdapter(

): PagingDataAdapter<String, NotificationAdapter.DataViewHolder>(diffCallback) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DataViewHolder {
        return DataViewHolder(
            ItemNotificationBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false)
        )
    }

    override fun onBindViewHolder(holder: DataViewHolder, position: Int) {
        getItem(position)?.let { holder.bindData(it) }
    }


    inner class DataViewHolder(val binding : ItemNotificationBinding) :
        RecyclerView.ViewHolder(binding.root) {

//        fun bindData(notification: NotificationDto) {
//
//            val type = notification.type
//            if(type==2) { // 임시 (각 type 별로 뭐가 '업데이트' 고 뭐가 '삭제' 고 뭐가 '추가' 인지 모름)
//                binding.icon.setBackgroundResource(R.drawable.ic_refresh)
//                binding.title.text = "업데이트" // R.string.notifications_noti_update 어떻게가져오지..
//            }
//
//            binding.date.text = notification.createdAt.substring(0, 10) // TODO : 'X시간 전' 으로 뜨는 건 직접 계산?
//            binding.description.text = notification.message
//        }
        fun bindData(notification: String) {
            binding.description.text = notification
        }

    }


//    companion object {
//        private val diffCallback = object : DiffUtil.ItemCallback<NotificationDto>() {
//            override fun areItemsTheSame(oldItem: NotificationDto, newItem: NotificationDto): Boolean {
//                return when (oldItem) {
//                    is NotificationDto -> (newItem as? NotificationDto)?.message == oldItem.message
//                    else -> false
//                }
//            }
//
//            override fun areContentsTheSame(oldItem: NotificationDto, newItem: NotificationDto): Boolean {
//                return oldItem == newItem
//            }
//        }
//    }
companion object {
    private val diffCallback = object : DiffUtil.ItemCallback<String>() {
        override fun areItemsTheSame(oldItem: String, newItem: String): Boolean {
            return when (oldItem) {
                is String -> (newItem as? String) == oldItem
                else -> false
            }
        }

        override fun areContentsTheSame(oldItem: String, newItem: String): Boolean {
            return oldItem == newItem
        }
    }
}

}
